Fixes Issue #24, added docs for Analytics Stack Reports.

Preeticp opened the pull request:
Added 3 modules for Stack Reports, made additions to the Glossary and added an assembly module for the same. Also, added a few supporting images.
Please note, while the concepts might remain the same, there are Major UX changes expected in the coming sprint, but I was asked to go ahead with the existing UX for now.

Preeticp commented:
@mishaone request you to review this PR. As mentioned, UX is in a flux, and going to change quite drastically, but this is as per the UX at the time of creating this doc and I have been asked to push this through for now. Concept level information would stay relevant in most cases. TIA

mishaone commented:
Sure. Will review this first thing tomorrow.

mishaone commented on docs/topics/modules/about_stack_reports.adoc:
```diff
@@ -0,0 +1,19 @@
+[#about_stack_reports]
+= About Stack Reports
+
+OSIO analyzes the health of your entire stack and its component libraries and provides recommendations to help you use relevant application components for your stack. It enables you to make informed decisions on the choice of open source components, augment your development stack, and reduce the risk to your organization.
```
There are a couple of instances of this, so just noting here that there are others that should be updated as well.

rkratky replied:
I see this hasn't been fixed yet. @Preeticp, are you planning to update this PR?

Preeticp replied:
Yes. On it.

Preeticp commented:
Hi Misha,
I have changed instances of passive voice to active. Used the linter tool where relevant. Also replaced OSIO with {osio}. Please have a look and ack if you are satisfied with the changes. I will then merge the PR.
Thank you.

rkratky commented:
@Preeticp, before you do any merging, please, make sure to build the doc on local; see the README.

rkratky commented:
Oops, didn't mean to close this. Clicked the wrong button. Reopened.

Preeticp commented:
ack @rkratky.
@mishaone are we good with this one? Please have a look and ack if you are satisfied with the changes.

mishaone commented:
Looks good to me, @Preeticp except a few instances of OSIO instead of {osio}. It's an ack from me. :)

rkratky commented on docs/topics/modules/interpreting_stack_report.adoc:
```diff
@@ -0,0 +1,47 @@
+The Stack Report comprises three components:
+
+== Stack report: An overall stack analysis
+The Stack report scans the total number of components in your stack, and reports those analyzed by {osio} and those unknown to it.
```
So, the feature is called "Stack [R|r]eport", and it scans and reports? And when it reports, it generates a report?

It's funny at best and confusing at all other times. If this description is correct, a bug should be filed against the name of the feature.

rkratky commented on docs/topics/modules/interpreting_stack_report.adoc:
```diff
@@ -0,0 +1,47 @@
+[#interpreting_stack_report]
+= Interpreting your Stack Report
+
+The Stack Report comprises three components:
```
I think the three components should be listed right here (as a bullet list) and link to the subsections.

rkratky commented on docs/topics/modules/interpreting_stack_report.adoc:
```diff
@@ -0,0 +1,47 @@
+
+It displays:
+
+Security Alerts:: {osio} examines the Common vulnerabilities and exposures (CVEs) for the components of the stack and highlights vulnerabilities, if any.
```
"{osio} examines"? I thought it was the Stack Report that does it...

rkratky commented on docs/topics/modules/interpreting_stack_report.adoc:
```diff
@@ -0,0 +1,47 @@
+It displays:
+
+Security Alerts:: {osio} examines the Common vulnerabilities and exposures (CVEs) for the components of the stack and highlights vulnerabilities, if any.
+Usage:: It analyzes your overall stack composition, compares it to other similar stacks and displays the number of components in your stack that are not commonly used in other similar stacks or that rarely go together.
```
"it analyzes"? What's "it"? OSIO, or Stack Report?

rkratky replied:
This problem with "it" vs "OSIO" vs something else is in this entire list. It should be consistent.

rkratky commented on docs/topics/modules/interpreting_stack_report.adoc:
```diff
@@ -0,0 +1,47 @@
+Usage:: It analyzes your overall stack composition, compares it to other similar stacks and displays the number of components in your stack that are not commonly used in other similar stacks or that rarely go together.
+Licenses:: It analyzes the licenses used in your stack, and displays the number of unknown licenses. It also displays the number of license conflicts at the stack level and the conflicting licenses used in the stack on mouse hover.
+
+== Detail analysis of your stack components
```
Detailed?

rkratky commented on docs/topics/modules/interpreting_stack_report.adoc:
```diff
@@ -0,0 +1,47 @@
+Licenses:: It analyzes the licenses used in your stack, and displays the number of unknown licenses. It also displays the number of license conflicts at the stack level and the conflicting licenses used in the stack on mouse hover.
+
+== Detail analysis of your stack components
+The detailed component analysis lists all the components in the stack, analyzes them on the basis of various parameters, and recommends relevant Alternate Components.
```
"detailed component analysis **lists** [...], **analyzes** [...], and **recommends**"

So, an anylysis analyzes?

rkratky commented on docs/topics/modules/interpreting_stack_report.adoc:
```diff
@@ -0,0 +1,47 @@
+== Detail analysis of your stack components
+The detailed component analysis lists all the components in the stack, analyzes them on the basis of various parameters, and recommends relevant Alternate Components.
+
+It helps you assess the credibility of your stack components by displaying component level details such as:
```
"It"? Is it OSIO, SR, or 'detailed analysis'?

rkratky commented on docs/topics/modules/interpreting_stack_report.adoc:
```diff
@@ -0,0 +1,47 @@
+
+It helps you assess the credibility of your stack components by displaying component level details such as:
+
+Versions:: It lists Current, Recommended, and the Latest available version of the components in your stack so that you can select a suitable version for your stack.
```
"Current, Recommended, and the Latest" -- why caps?

rkratky replied:
"stack so that" -> "stack, so that"

rkratky commented on docs/topics/modules/interpreting_stack_report.adoc:
```diff
@@ -0,0 +1,47 @@
+
+Security Issue:: Based on the CVEs of the stack components, {osio} identifies security vulnerabilities in your stack components. On mouse hover, it displays the CVE scores of components identified with security vulnerabilities.
+
+License:: It lists Licenses for each component, flags Unknown and Conflicting licenses, and recommends *Alternate Components* in case of *License Outliers*.
```
Why capitals?

rkratky commented on docs/topics/modules/interpreting_stack_report.adoc:
```diff
@@ -0,0 +1,47 @@
+
+OSIO Usage:: It tracks the number of projects in {osio} that use the same component. It recommends *Alternate Components* in case of *Usage Outliers*.
+
+GitHub Statistics:: {osio} tracks GitHub statistics such as Usage and number of Watchers for the components of your stack.
```
Usage and Watchers should be marked-up.

statistics such as -> statistics, such as

rkratky commented on docs/topics/modules/using_stack_report.adoc:
```diff
@@ -0,0 +1,62 @@
+[#using_stack_report]
+= Using your Stack Report
+
+{osio} Analytics helps you make informed choices with respect to your development stack and its component libraries by analyzing your stack at the development stage, within your IDE, and at the deployment stage.
```
So, it's "Analytics", not "SR"?

Preeticp replied:
Yes, I have made changes accordingly.

rkratky commented on docs/topics/modules/using_stack_report.adoc:
```diff
@@ -0,0 +1,62 @@
+
+.Procedure
+
+* In the *Stack Report Recommendations* section of your space dashboard, click *Show complete stack report* to access the detailed stack report for your manifest files.
```
The capitalization problem is in the app, too.

Preeticp replied:
Yes, there are a lot of inconsistencies wrt cases, US>UK english. I have pointed them out. Also, UX is changing, so a lot of these labels may just disappear. I have however changed references to stack reports to all small in the doc, except when it  refers to a field in UX or is part of a Title/Heading.

rkratky commented on docs/topics/modules/using_stack_report.adoc:
```diff
@@ -0,0 +1,62 @@
+
+Alternatively, you can access your stack report as follows:
+
+. Click *Create*, and then *Pipelines* to view the pipeline builds for your project.
```
Remove comma.

rkratky commented on docs/topics/modules/using_stack_report.adoc:
```diff
@@ -0,0 +1,62 @@
+
+* *Stack Report:* A synopsis of the stack's analysis report
+
+* *Detail analysis of your stack components:* A list of all the stack components and their various parameters, including recommendations of Alternate Components in case of License Outliers or Usage Outliers.
```
Detailed?

rkratky commented on docs/topics/modules/using_stack_report.adoc:
```diff
@@ -0,0 +1,62 @@
+
+* *Detail analysis of your stack components:* A list of all the stack components and their various parameters, including recommendations of Alternate Components in case of License Outliers or Usage Outliers.
+
+* *Additional components recommended by {osio}:* A list of additional components, commonly used by other similar stacks.
```
Remove comma.
Remove 'other'.

rkratky commented on docs/topics/modules/using_stack_report.adoc:
```diff
@@ -0,0 +1,62 @@
++
+See <<interpreting_stack_report, Interpreting your Stack Report>> section for further details.
+
+. Filter the *Detail analysis of your stack components* list to see:
```
Detailed?

rkratky commented on docs/topics/modules/using_stack_report.adoc:
```diff
@@ -0,0 +1,62 @@
+** *<<glossary.adoc#Unknown Licenses, Unknown Licenses>>* (image:license_unknown.png[title="Unknown Licenses"])
+** *<<glossary.adoc#License conflict, License conflict in a Component>>* (image:license_conflicts.png[title="License Conflicts"])
++
+Note that, {osio} recommends *Alternate Components* for your stack if it identifies License or Usage outliers.
```
Remove comma.

Preeticp commented on docs/topics/modules/glossary.adoc:
```diff
@@ -16,3 +16,15 @@ Pipeline:: Pipelines are a continuous delivery system that, at each step, tests
+
+License outliers:: Licenses which rarely go together or not commonly used in other similar stacks.
+
+Alternate Components:: Components recommended by {osio}, to replace License or Usage Outliers.
```
Alternate components are recommended in case of Usage and License Outliers in the Stack Report and in Che in the case of Security issues, hence listed it below those three components. Also this follows the UX order of filters in the detailed analysis of the table. Are you saying this cause it is a recommendation and the rest are observations based on the data?

rkratky replied:
> Are you suggesting this because it is a recommendation and the rest are observations based on the data?

Yes.

Preeticp commented:
@rkratky, I have addressed all the comments and made changes accordingly, please check and ack or let me know if anything else needs to be looked into. Thank you.
